ksql-test-commons/src/test/java/io/confluent/ksql/test/commons/TestCaseTest.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.test.commons;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+import io.confluent.ksql.metastore.model.KsqlTopic;
+import io.confluent.ksql.serde.json.KsqlJsonTopicSerDe;
+import io.confluent.ksql.util.KafkaTopicClient;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Optional;
+import org.apache.avro.Schema;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.streams.TopologyTestDriver;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TestCaseTest {
+
+  @Rule
+  public final org.junit.rules.ExpectedException expectedException = org.junit.rules.ExpectedException
```

Comment:
nit: maybe rename `ExpectedException` to `KsqlExceptedException` so that we don't accidentally import it every time we want ExpectedException in tests?

Reply:
I agree with your suggestion but it would be a big change since we have to update existing QTT files. I would delay it to the PR that will require such changes when we update the test file format as being discussed in the KLIP.

Reply:
No QTT tests would need to change. Just change the class name - not the name of the JSON field name.